### ISSUE TITLE
Click on notification doesn't redirect to the right tab (team)

### ISSUE THREAD

**Issue by guyguy333** [mattermost/desktop#321]
- [X] This is not a feature idea. Feature ideas go here: http://www.mattermost.org/feature-ideas/
- [X] This is not a troubleshooting question. Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/
- [X] This is a product defect issue that can be reproduced using the steps listed below.
- [X] I have read [CONTRIBUTING.md](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md). 

---
#### Summary

When you get a notification with a new message and you click on it, instead of switching to the right tab (team) in the desktop app, it changes the team in the previously active team (tab).
#### Steps to reproduce

Operating System: OS X 10.11.6
Mattermost Desktop App version: 3.4.0
Mattermost Server version: 3.4.0

Adds two teams to Mattermost desktop. Each team appears in a different tab in the desktop app. Select a tab and wait a notification from the other team. Click on it. The tab will not change, it will instead switch team in the current tab.
#### Expected behavior

When you click on the notification, it should switch tab focus instead of changing team in the current tab.
#### Observed behavior

Currently, it changes team in the tab (leading to two tabs with the same team) who you click on the notification.

**Comment by jasonblais** [comment]
Not reproduced on Windows 10, known issue [in Windows 7](https://github.com/mattermost/desktop/issues/67).

I thought this behaviour used to work for OS X though, so adding a bug label for it,

**Comment by yuya-oc** [comment]
@guyguy333 To make sure, would you check whether you are using the latest release? v3.4.0-rc1 had a bug which is related to notifications. If you are using correct version, please show console output when clicking notifications (`Alt+Cmd+I` shows the console).

**Comment by St-Ex** [comment]
Same problem on linux Mint 18. When clicking on notification, current tab switch to team/channel instead of relevant tab switching to channel.

**Comment by jasonblais** [comment]
Did it use to work for you @St-Ex and @guyguy333, or is this a new problem with Desktop App v3.4?

Also, if you're able to provide a console output, that would be great: 
 1) In the Mattermost Desktop App, open developer console `Ctrl+Shift+I` on Windows/Linux, `Alt+Cmd+I` on Mac)
 2) Choose `Console` on the developer console

![image](https://cloud.githubusercontent.com/assets/13119842/18953785/146d3030-861e-11e6-9b80-2bf43c0d096e.png)

 3) Switch to [Team1] on the Desktop App
 4) Switch to another app/browser on your machine 
 5) Receive a message from someone in [Team2]
 6) Click on the desktop notification

This should print messages on the developer console, which will make it easier for us to troubleshoot the problems.

**Comment by St-Ex** [comment]
I don't remember this ever working. 

`index.js:526 [B-APPLI] method=getChannel msg=il n'existe pas de canal dont l'identifiant est =fm9k9rsg8ir43emj9deja58her pour l'équipe dont le nom est=dm5riffwktnxdexq8pdp4dhxgy detail= rid=qmkswsarxfd1bddiqsnjwocuoewebview.addEventListener.e @ index.js:526WebViewImpl.dispatchEvent @ /opt/Mattermost/resources/electron.asar/renderer/web-view/web-view.js:192dispatchEvent @ /opt/Mattermost/resources/electron.asar/renderer/web-view/guest-view-internal.js:57(anonymous function) @ /opt/Mattermost/resources/electron.asar/renderer/web-view/guest-view-internal.js:66emitMany @ events.js:127emit @ events.js:201
index.js:526 [B-APPLI] [object Object]webview.addEventListener.e @ index.js:526WebViewImpl.dispatchEvent @ /opt/Mattermost/resources/electron.asar/renderer/web-view/web-view.js:192dispatchEvent @ /opt/Mattermost/resources/electron.asar/renderer/web-view/guest-view-internal.js:57(anonymous function) @ /opt/Mattermost/resources/electron.asar/renderer/web-view/guest-view-internal.js:66emitMany @ events.js:127emit @ events.js:201
index.js:526 [SFAM] method=getChannel msg=il n'existe pas de canal dont l'identifiant est =fm9k9rsg8ir43emj9deja58her pour l'équipe dont le nom est=dm5riffwktnxdexq8pdp4dhxgy detail= rid=3hti9r5pw7dyuct1jhyyiqqibowebview.addEventListener.e @ index.js:526WebViewImpl.dispatchEvent @ /opt/Mattermost/resources/electron.asar/renderer/web-view/web-view.js:192dispatchEvent @ /opt/Mattermost/resources/electron.asar/renderer/web-view/guest-view-internal.js:57(anonymous function) @ /opt/Mattermost/resources/electron.asar/renderer/web-view/guest-view-internal.js:66emitMany @ events.js:127emit @ events.js:201
index.js:526 [SFAM] [object Object]webview.addEventListener.e @ index.js:526WebViewImpl.dispatchEvent @ /opt/Mattermost/resources/electron.asar/renderer/web-view/web-view.js:192dispatchEvent @ /opt/Mattermost/resources/electron.asar/renderer/web-view/guest-view-internal.js:57(anonymous function) @ /opt/Mattermost/resources/electron.asar/renderer/web-view/guest-view-internal.js:66emitMany @ events.js:127emit @ events.js:201`

**Comment by St-Ex** [comment]
I think the problem is on index.jsx

```js
var handleNotificationClick = function() {
        thisObj.handleSelect(index);
      }
```

We do not look for the tab to select. However, it will be hard because we have to change the notification onclick event to add the notification url but we don't have it. It's inside the chorme notification itself.

**Comment by yuya-oc** [comment]
Hmm...the feature works fine in my Mac.

`index.jsx` is surely complicated. `index` is passed from `prop.teams.map()`, so it should open correct tab when it's called. For this purpose, `Notification.onclick` has been already changed in `src/browser/webview/mattermost.js` and `src/browser/js/notification.js`. It works like below.

1. Click a notification.
2. Call modified onclick event.
3. The event calls `handleNotificationClick()` via some callbacks.
4. Finally, call the original onlick event which is set by mattermost webapp.

Please let me confirm, does the two teams exist on the same server? Or different servers? In my app, there are two teams:

1. team1 on server1
2. team2 on server2

**Comment by St-Ex** [comment]
Both on same server. 

I just discovered something. I add some logs after notification :
```js
notification: function(title, options) {
    ipc.send('notified', {
      title: title,
      options: options
    });
    console.dir('notification-log',this)
  },
```
I can clearly see 2 notifications, on per tab being log
index.js:520 [SFAM] notification-log
index.js:520 [B-APPLI] notification-log

**Comment by St-Ex** [comment]
Bon, le problème est exactement le même avec 2 fenêtres firefox ouvertes sur 2 team.

**Comment by St-Ex** [comment]
https://github.com/mattermost/platform/issues/4116

**Comment by jasonblais** [comment]
Much appreciated, thanks @St-Ex!

**Comment by jasonblais** [comment]
Jira ticket was created by the Mattermost team to fix on the server side, adding here for us to track: https://mattermost.atlassian.net/browse/PLT-4334

**Comment by jasonblais** [comment]
@guyguy333 This issue is resolved in Mattermost Server 3.6, which will release on January 16th.

Thanks for all your help @St-Ex !
